2003-09-10  H.J. Lu  <hongjiu.lu@intel.com>

	* elfxx-ia64.c (elfNN_ia64_special_sections): Mark all sections
	started with ".sbss" or "".sdata" as SHF_IA_64_SHORT.

2003-08-22  H.J. Lu  <hongjiu.lu@intel.com>

	* elfxx-ia64.c (elfNN_ia64_create_dynamic_sections): Align the
	.got section at 8 bytes.

--- bfd/elfxx-ia64.c	20 Aug 2003 08:37:18 -0000	1.98
+++ bfd/elfxx-ia64.c	10 Sep 2003 20:48:32 -0000	1.100
@@ -1781,6 +1781,8 @@ elfNN_ia64_create_dynamic_sections (abfd
   {
     flagword flags = bfd_get_section_flags (abfd, ia64_info->got_sec);
     bfd_set_section_flags (abfd, ia64_info->got_sec, SEC_SMALL_DATA | flags);
+    /* The .got section is always aligned at 8 bytes.  */
+    bfd_set_section_alignment (abfd, ia64_info->got_sec, 3);
   }
 
   if (!get_pltoff (abfd, info, ia64_info))
@@ -4746,9 +4748,9 @@ elfNN_ia64_reloc_type_class (rela)
 
 static struct bfd_elf_special_section const elfNN_ia64_special_sections[]=
 {
-  { ".sbss",		0,	NULL,	0,
+  { ".sbss",		5,	NULL,	0,
     SHT_NOBITS,		SHF_ALLOC + SHF_WRITE + SHF_IA_64_SHORT },
-  { ".sdata",		0,	NULL,	0,
+  { ".sdata",		6,	NULL,	0,
     SHT_PROGBITS,	SHF_ALLOC + SHF_WRITE + SHF_IA_64_SHORT },
   { NULL,		0,	NULL,	0,
     0,			0 }
